toolchain/check/check.cpp

```diff
@@ -171,14 +198,98 @@ static auto ImportCurrentPackage(Context& context, UnitInfo& unit_info,
 
   ImportLibrariesFromCurrentPackage(
       context, namespace_type_id,
-      CollectTransitiveImports(self_import, total_ir_count));
+      CollectTransitiveImports(&self_import, /*api_imports=*/nullptr,
+                               total_ir_count));
 
   context.scope_stack().Push(
       package_inst_id, SemIR::NameScopeId::Package,
       context.name_scopes().Get(SemIR::NameScopeId::Package).has_error);
 }
 
+// Imports all other packages (excluding the current package).
+static auto ImportOtherPackages(Context& context, UnitInfo& unit_info,
+                                int total_ir_count,
+                                SemIR::TypeId namespace_type_id) -> void {
+  // For each package, the index into the API's package_imports. If this is not
+  // imported by the impl, the IdentifierId is the name of the package being
+  // imported. This is initially the size of the implementation's imports, even
+  // when there is no API, for simplicity in iteration.
```

Comment:
If my understanding of the code below is correct, then:
```suggestion
  // For each package, the index into the API's package_imports. If this package is
  // imported by the API file, the IdentifierId is the name of the package being
  // imported; otherwise, it is invalid and the index is -1. This is initially the size of the implementation's imports, even
  // when there is no API, for simplicity in iteration.
```

Reply:
Also, "even when there is no API" might be clearer as "even when this file is the API file" or something?

Reply:
Rewriting this comment, trying to incorporate suggestions.